copycat/file/src/main/java/org/apache/kafka/copycat/file/FileStreamSourceTask.java

```diff
@@ -193,4 +190,12 @@ public void stop() {
             this.notify();
         }
     }
+
+    private Map<String, String> offsetKey(String filename) {
```

Comment:
These are connector-specific, there's not a ton we can do to simplify/share this since `Collections.singletonMap` already made these methods quite concise. It's a good idea for connectors to have methods like these since they ensure all cases where an offset key or value is constructed are consistent.

I think the only thing we might be able to provide to help with this is something like `MapBuilder` so each connectors `offsetKey` and `offsetValue` method can be simplified by chaining, e.g. something like:
```
private Map<String, Object> offsetKey(String namespace, String filename) {
    return MapBuilder.builder().put(NAMESPACE_FIELD, namespace).put(FILENAME_FIELD, filename).build();
}
```
That's not at all critical though. It's just a very slightly nicer API, so I don't think it's all that important to include in this patch.